controller/contact.js

```diff
@@ -22,16 +17,56 @@ const schema = Joi.object({
 });
 
 const get = async (req, res, next) => {
+  const { page, limit, filter } = req.query;
+  const parsedPage = parseInt(page);
+  const parsedLimit = parseInt(limit);
+  const parsedFilter = filter === "true";
+
   try {
     const contacts = await service.getContacts();
-    res.json(genereteJSON("success", 200, "contacts", contacts));
+
+    // Pagination for query params: Page and Limit
+    const endlyIndex = parsedPage * parsedLimit;
+    const startlyIndex = parsedPage === 1 ? 0 : endlyIndex - parsedLimit;
+
+    const contactsOnChosenPage = contacts.splice(startlyIndex, parsedLimit);
+
+    if (page && limit) {
+      return res.json(
+        genereteJSON("success", 200, "contacts", {
+          user: req.user,
+          contacts: contactsOnChosenPage,
+        }),
+      );
+    }
+
+    // Pagination for query param: Filter
+    const contactsFilteredByFavorite = contacts.filter(
+      (c) => c.favorite === parsedFilter,
+    );
+
+    if (filter) {
+      return res.json(
+        genereteJSON("success", 200, "contacts", {
+          user: req.user,
+          contacts: contactsFilteredByFavorite,
+        }),
+      );
+    }
+
+    res.json(
+      genereteJSON("success", 200, "contacts", {
+        user: req.user,
+        contacts,
+      }),
+    );
   } catch (error) {
     console.log(error);
     next(error);
   }
 };
 
-const getById = async (req, res, next) => {
```

Comment:
Pamiętaj że od tej pory potrzebujesz sprawdzać po kluczu owner, czy  kontakt należy do użytkownika. Dotyczy to operacji na kontaktach: get, post, put, delete.

Reply:
Od tej pory metody muszą zawierać id użytkownika, który dokonuje operacji tzw owner.

Reply:
+